### PR TITLE
Add some corner case tests for SPDX, and fix a discovered issue

### PR DESCRIPTION
**Pull request by ctron**
This fixes the two issues of https://github.com/trustification/trustify/discussions/555 but removing duplicated during the conversion and by renaming the enum variant to `All`, which better reflects what it currently does.

### CODE REVIEW

File: modules/fundamental/src/sbom/service/sbom.rs [hunk 1]
```diff
@@ -460,7 +460,19 @@ impl SbomService {
             )
             .await?;
 
-        Ok(result.items.into_iter().map(|r| r.package).collect())
+        // TODO: this will break when adding pagination, as we effectively only process a single page
```
Comment (bobmcwhirter) [inline modules/fundamental/src/sbom/service/sbom.rs:463]: can we leverage some `DISTINCT` on the query?

**Review by jcrossley3 — APPROVED** [review]

File: modules/fundamental/src/sbom/service/sbom.rs (hunk 1 above)
Comment (ctron) [reply modules/fundamental/src/sbom/service/sbom.rs:463]: Once we need this, we'd need to re-implement the query to return only packages, instead of full package-relationships. Currently, this re-uses the "fetch package relationships" function, stripping away the unnecessary information.
